models/demos/falcon40b/demo/demo.py

```diff
@@ -209,17 +209,12 @@ def run_falcon_demo_kv(
     # State dict is needed for embeddings
     logger.info("Loading weights...")
     profiler.start(f"loading_weights")
-    if (tt_cache_path == Path(f"models/demos/falcon40b/datasets/{model_version}")) and (
-        len(os.listdir(f"models/demos/falcon40b/datasets/{model_version}")) < 260
-    ):
```

Comment:
The problem is also that the ttnn way of loading a cached tensor expects a tensor, thus passing in None does not work. We could write a workaround or load the embeddings manually with ttlib as in the other modules but I think sooner or later we'll want to switch to ttnn for all modules anyway. I guess the question is whether we need this to work without ever loading the ref model, because it doesn't take that long to load it once it's downloaded and cached, but if it's not downloaded it would be very unnecessary to do so, yes.